solvebio/query.py

```diff
@@ -880,26 +904,38 @@ def join(self, query_b, key, key_b=None, prefix="b_", always_prefix=True):
             else:
                 name = field.name
 
+            query_b_prefixed_fields.append(name)
+
             if name in existing_field_names:
-                logger.warning("Field '{}' will be overwritten in the join results")
+                logger.warning("Field '{}' will be overwritten in the join results".format(name))
 
             target_fields.append({
                 "name": name,
                 "title": field.title,
                 "data_type": field.data_type,
                 "ordering": field.ordering,
-                "is_list": False,
+                "is_list": True,
                 "is_transient": False,
                 "expression": """
-                    get(record, "{}.{}")
+                    [get(item, "{}.{}") for item in record.exploded_join_field]
                 """.format(query_b_join_field_name, field.name),
-                "depends_on": [query_b_join_field_name]
+                "depends_on": [query_b_join_field_name, 'exploded_join_field']
             })
 
         # Add to any existing target fields
         new_query._target_fields += target_fields
 
-        return new_query
+        # Explode new_query records
```

Comment:
the post_annotation_expression should be passed into the new_query object instead of annotating the output.

Reply:
Done.